common/story_cycles/story_cycle_animal_companion.txt

```diff
@@ -0,0 +1,279 @@
+					limit = {
+						NOT = {
+							any_owned_story = {
+									story_type = story_cycle_animal_companion
+								}
+						}
+						NOT = { is_in_list = can_get_companion }
+						NOR = {
+							has_relation_rival = scope:story.story_owner
+							has_relation_potential_rival = scope:story.story_owner
+						}
+					}
+					add_to_list = can_get_companion
+				}
+			}
+			if = {
+				limit = {
+					any_in_list = {
+						list = can_get_companion
+						count >= 1
+					}
+					exists = scope:story #There seems to be instances where the story is already gone
+					exists = scope:story.var:story_cycle_cat_name
+				}
+				random_in_list = {
+					list = can_get_companion
+					weight = {
+						base = 10
+						modifier = {
+							add = 100
+							is_player_heir_of_trigger = { CHARACTER = scope:story.story_owner }
+						}
+						modifier = {
+							add = 50
+							is_child_of = scope:story.story_owner
+						}
+					}
+					save_scope_as = person_inheriting
+					scope:story = {
+						set_variable = {
+							name = original_owner
+							value = scope:story.story_owner
+						}
+					}
+					trigger_event = pet_animal.1198
+				}
+				transfer_companion_story_cycle_to_effect = {
+					CHARACTER = scope:person_inheriting
+					STORY = scope:story
+				}
+			}
+			else = {
+				scope:story = { end_story = yes }
+			}
+		}
+	}
+
+	# Give the companion a name!
+	effect_group = {
+		days = { 2 4 }
+		chance = 100
+	
+		triggered_effect = {
+			trigger = {
+				story_owner = {
+					NOT = { has_character_flag = is_naming_companion }
+				}
+				NOT = { exists = story_owner.var:story_cycle_animal_companion_name }
+			}
+			effect = {
+				story_owner = {
+					add_character_flag = is_naming_companion
+					trigger_event = companion_story_cycle.1001
+				}
+			}
+		}
+	}
+
+	#To save the name in the story's scope (can for some reason not be done in the naming event itself)
+	effect_group = {
+		days = 5
+		chance = 100
+	
+		triggered_effect = {
+			trigger = {
+				exists = story_owner.var:story_cycle_animal_companion_name
+			}
+			effect = {
+				set_variable = {
+					name = story_cycle_animal_companion_name
+					value = scope:story.story_owner.var:story_cycle_animal_companion_name
+				}
+			}
+		}
+	}
+
+	#Age the companion
+	effect_group = {
+		days = 365
+
+		trigger = {
+			exists = var:companion_age_variable
+		}
+
+		triggered_effect = {
+			trigger = { always = yes }
+			effect = {
+				change_variable = {
+					name = companion_age_variable
+					add = 1
+				}
+			}
+		}
+	}
+
+	# The companion dies :(
+	effect_group = {
+		days = { 30 60 }
+		chance = 75
+
+		trigger = {
+			var:companion_age_variable >= var:companion_lifespan_variable
+		}
+	
+		triggered_effect = {
+			trigger = {
+				exists = story_owner.var:story_cycle_animal_companion_name
+			}
+			effect = {
+				story_owner = {
+					add_character_flag = {
+						flag = companion_is_dying
+					}
+					trigger_event = companion_story_cycle.1199
+				}
+			}
+		}
+	}
```

Comment:
So... I guess you can make dying in the "aging" `effect_group` instead to save place.

Reply:
Not sure I want that since that would mean that the companion can only die on their birthday